src/miner.cpp

```diff
@@ -344,7 +344,8 @@ class AddOutputsToCoinbaseTxAndSign
 
 CMutableTransaction CreateCoinbaseTransaction(const CChainParams& chainparams, CAmount nFees, const MinerAddress& minerAddress, int nHeight)
 {
-        CMutableTransaction mtx = CreateNewContextualCMutableTransaction(chainparams.GetConsensus(), nHeight);
+        CMutableTransaction mtx = CreateNewContextualCMutableTransaction(
+                chainparams.GetConsensus(), nHeight, nPreferredTxVersion < ZIP225_MIN_TX_VERSION);
```

Comment:
This breaks when `minerAddress` has resolved to an Orchard receiver. We need to check whether it holds `OrchardRawAddress` and ignore the flag in this case.

@nuttycom notes that this might be why he had to set `-preferredtxversion=5` in some tests. If that's the case, then fixing this should enable that flag to be removed from those tests.